### PR TITLE
Avoid unconditionally exporting stack functions

### PR DESCRIPTION
**Pull request by sbc100**
See #11898

### CODE REVIEW

**Review by dschuff — APPROVED** [review]

File: tools/link.py [hunk 1]
```diff
@@ -1279,7 +1279,13 @@ def phase_linker_setup(options, state, newargs):
     if sym in settings.EXPORTED_RUNTIME_METHODS:
       settings.REQUIRED_EXPORTS.append(sym)
 
-  settings.REQUIRED_EXPORTS += ['stackSave', 'stackRestore', 'stackAlloc']
+  if settings.MAIN_READS_PARAMS and not settings.STANDALONE_WASM:
```
Comment (kripken) [inline tools/link.py:1282]: Don't we still need this in standalone wasm mode? We still emit the JS in that case even if the wasm is meant to be standalone, and that JS uses `stackAlloc`. But perhaps we can move the arguments logic into the entry function somehow?

Comment (sbc100) [reply tools/link.py:1282]: In standalone wasm mode `callMain` calls `_start` which takes zero argument, and we don't call stackAlloc.

Comment (kripken) [reply tools/link.py:1282]: Thanks, that does sound familiar. Then can we ifdef around the `stackAlloc`-using code then? I don't care as a code size optimization (as it's for standalone mode) but closure and other tools might error if we have a call but no function to call.

lgtm though, again, as that's an issue for JS in standalone mode.

Comment (sbc100) [reply tools/link.py:1282]: There is already an `#ifdef` around the use of `staticAlloc` in `callMain`.  See postable.js line 66

Comment (kripken) [reply tools/link.py:1282]: Ah, you're very right, I was misreading that code. Sorry!

**Review by kripken — APPROVED** [review]

**Pull request merged**